messaging/src/main/java/org/axonframework/messaging/Message.java

```diff
@@ -75,9 +75,18 @@ public interface Message<T> extends Serializable {
      * @return the type of payload.
      * @deprecated Payloads are just jvm-internal representations. No need for matching against payload types
      */
-    @Deprecated // TODO #3085 - Introduce QualifiedName
+    @Deprecated // TODO #3085 - Replace for getMessageType once fully integrated
     Class<T> getPayloadType();
 
+    /**
+     * Returns the {@link QualifiedName message type} of this {@link Message}.
+     *
+     * @return The {@link QualifiedName message type} of this {@link Message}.
+     */
+    default QualifiedName getMessageType() {
+        return QualifiedName.fromClass(getPayloadType());
```

Comment:
I think we should remove the default implementation. With this default, converting a message from one payload type to another, the qualified name also changes.

Reply:
I was trying to keep this pull request small...although I do agree. Let me give it a try to **enlarge** the amount of changes in this PR. I'll keep you posted :+1: